nodePort paramater missing when using service_type: nodeport

srinathman opened the pull request:
It seems that the "nodeport_port" is not actually setting nodeport to the desired static value, instead a random port is assigned. 
As per https://kubernetes.io/docs/concepts/services-networking/service/, a "nodePort" parameter must be provided for static nodeport_port, otherwise k8s will pick up a random port.
Just setting "port" and "targetPort" is not enough.

srinathman committed:
Update service.yaml.j2

As per https://kubernetes.io/docs/concepts/services-networking/service/, a "nodeport" parameter must be provided for static nodeport, otherwise k8s will pick up a random port.
Just setting "port" and "targetPort" is not enough

shanemcd commented:
After reading the docs you linked to... shouldn't `port: {{ nodeport_port }}` get changed to `port: 8052` ?

srinathman commented:
I wasn't sure why the original code was like, but yes, I believe it should be 8052

srinathman committed:
Merge branch 'ansible:devel' into patch-1

srinathman committed:
nodePort paramater missing when using nodeport

It seems that the "nodeport_port" is not actually setting nodeport to the desired static value, instead a random port is assigned.
As per https://kubernetes.io/docs/concepts/services-networking/service/, a "nodePort" parameter must be provided for static nodeport_port, otherwise k8s will pick up a random port.
Just setting "port" and "targetPort" is not enough.

shanemcd commented:
Sorry for the delay here, this slipped through the cracks. I see there is a conflict now so I'm going to close this in favor of https://github.com/ansible/awx-operator/pull/789.
